doc/build-helpers/fixed-point-arguments.chapter.md

```diff
@@ -0,0 +1,84 @@
+
+## Defining a build helper with `lib.extendMkDerivation` {#sec-build-helper-extendMkDerivation}
+
+The Nixpkgs library function [`lib.customisation.extendMkDerivation`](#function-library-lib.customisation.extendMkDerivation) can be used to define a build helper supporting fixed-point arguments based on another build helper with such support. As most of the build helpers are based on `stdenv.mkDerivation`, It can also bring fixed-point argument support to existing build helpers as a wrapper around the original definition.
+
+The following is the conventional definition of build helper `mkLocalDerivation` that doesn't support fixed-point arguments:
+
+:::{.example #ex-build-helpers-mkLocalDerivation-plain}
+
+# Example definition of `mkLocalDerivation` without fixed-point arguments support
+
+Write the followings in `pkgs/by-name/mk/mkLocalDerivation/package.nix`:
+
+```nix
+# Dependencies to be injected with `pkg.callPackage`
+{ lib
+, stdenv
+}:
+
+# Expression that evaulates to the build helper
+{ preferLocalBuild ? true
+, allowSubstitute ? false
+, impassablePredicate ? (_: false)
+, ...
+}@args:
+
+stdenv.mkDerivation (removeAttrs args [
+  "impassablePredicate"
+] // {
+  inherit
+    preferLocalBuild
+    allowSubstitute
+    ;
+})
+```
+
+:::
+
+A wrapper of `lib.extendMkDerivation` around the original definition brings the magical `finalAttrs` to the build helper:
+
+:::{.example #ex-build-helpers-mkLocalDerivation-extendMkDerivation}
+
+# Example definition of `mkLocalDerivation` extended from `stdenv.mkDerivation` with `lib.extendMkDerivation`
+
+Write in `pkgs/by-name/mk/mkLocalDerivation/package.nix`:
+
+```nix
+# Dependencies to be injected with `pkg.callPackage`
+{ lib
+, stdenv
+}:
+
+# Expression that evaulates to the build helper
+lib.extendMkDerivation stdenv.mkDerivation (finalAttrs:
+
+{ preferLocalBuild ? true
+, allowSubstitute ? false
+, impassablePredicate ? (_: false)
+, ...
+}@args:
+
+removeAttrs args [
+  "impassablePredicate"
+] // {
+  inherit
+    preferLocalBuild
+    allowSubstitute
+    ;
+})
+```
+
+:::
+
+Should there be a need to modify the result derivation, `lib.customisation.extendMkDerivationModified` additionally takes a derivation modification function as the first argument. This comes in handy when we would like to add some default overridings to the result derivation, or decorate it with [`lib.extendDerivation`](#function-library-lib.customisation.extendDerivation).
+
+Extending a build helper with `lib.extendMkDerivation` is different from extending the fixed-point arguments with `lib.extend` or [`<pkg>.overrideAttrs`](#sec-pkg-overrideAttrs), as the implementation can choose *not* to pass some input arguments (e.g. `impassablePredicate`) down the base build helper instead of having to pass everything like a conventional overriding/overlay.
```

Comment:
```suggestion
Extending a build helper with `lib.extendMkDerivation` is different from extending the fixed-point arguments with `lib.extend` or [`<pkg>.overrideAttrs`](#sec-pkg-overrideAttrs), as the implementation can choose *not* to pass some input arguments (e.g. `impassablePredicate`) down the base build helper instead of having to pass everything like a conventional override.
```

Not sure if overlay would even be the right thing to mention here. In any case, it would help to link to the relevant section, even if one can expect readers to know that one already - for reference, to look up how these concepts relate.